com.unity.shadergraph/Editor/Drawing/Inspector/SGPropertyDrawer.cs

```diff
@@ -0,0 +1,1224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using UnityEditor;
+using UnityEditor.Graphing.Util;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEditor.UIElements;
+using UnityEngine;
+using UnityEngine.UIElements;
+using FloatField = UnityEditor.ShaderGraph.Drawing.FloatField;
+
+namespace Drawing.Inspector
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public class SGPropertyDrawer : Attribute
+    {
+        public Type propertyType { get; private set; }
+
+        public SGPropertyDrawer(Type propertyType)
+        {
+            this.propertyType = propertyType;
+        }
+    }
+
+    // Interface that should be implemented by any property drawer for the inspector view
+    public interface IPropertyDrawer
+    {
+        Action inspectorUpdateDelegate { get; set; }
+
+        VisualElement DrawProperty(PropertyInfo propertyInfo, object actualObject, InspectableAttribute attribute);
+    }
+
+    [SGPropertyDrawer(typeof(Enum))]
+    class EnumPropertyDrawer : IPropertyDrawer
```

Comment:
Please spit out each specific class into its own file (in this same folder is fine).

Reply:
All property drawers are in their own separate files now under: `Drawing/Inspector/PropertyDrawers`